### PR TITLE
Create NavigationItem view.

### PR DESCRIPTION
**Pull request by RomanEsin**

### CODE REVIEW

**Comment by RomanEsin** [comment]
This is basically UINavigationItem but in SwiftUI.

File: Project SF/Views/NavigationItem.swift [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+import SwiftUI
+
+struct NavigationItem: View {
+    let imageName: String
+    let action: () -> Void
+
+    var body: some View {
+        Button(action: action, label: {
+            Image(systemName: "plus")
```
Comment (Priva28) [inline Project SF/Views/NavigationItem.swift:16]: Don't you need to set this to be the imageName?

File: Project SF/Views/NavigationItem.swift [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    init(imageName: String, action: @escaping () -> Void = {}) {
+        self.imageName = imageName
+        self.action = action
+    }
+}
+
+struct NavigationItem_Previews: PreviewProvider {
+    static var previews: some View {
+        NavigationItem()
```
Comment (Priva28) [inline Project SF/Views/NavigationItem.swift:28]: Does this compile? Image name which doesn't have a default value isn't specified.

**Review by Priva28 — CHANGES_REQUESTED** [review]
Looks good but is this necessary? It's really just a button with an image as it's label which doesn't normally require too much code.

File: Project SF/Views/NavigationItem.swift (hunk 1 above)
Comment (RomanEsin) [reply Project SF/Views/NavigationItem.swift:16]: Oh true, will fix a bit later

File: Project SF/Views/NavigationItem.swift (hunk 2 above)
Comment (RomanEsin) [reply Project SF/Views/NavigationItem.swift:28]: Oh I’m dumb, forgot to build it. Interesting how build action actually passes

**Comment by RomanEsin** [comment]
> Looks good but is this necessary? It's really just a button with an image as it's label which doesn't normally require too much code.

True, but this looks much more cleaner in code.

**Comment by Priva28** [comment]
That's fine. Just check those 2 things I commented and it's fine to merge.

**Review by wfltaylor — APPROVED** [review]

**Pull request merged**